checkov/terraform/runner.py

```diff
@@ -119,7 +121,34 @@ def run_block(self, entities,
             entity_evaluations = None
             context_parser = parser_registry.context_parsers[block_type]
             definition_path = context_parser.get_entity_context_path(entity)
-            entity_id = ".".join(definition_path)
+            entity_id = ".".join(definition_path)       # example: aws_s3_bucket.my_bucket
+
+            caller_file_path = None
+            caller_file_line_range = None
+
+            if module_referrer is not None:
+                referrer_id = self._find_id_for_referrer(full_file_path,
+                                                         self.tf_definitions)
+                if referrer_id:
+                    entity_id = f"{referrer_id}.{entity_id}"        # ex: module.my_module.aws_s3_bucket.my_bucket
+                    abs_caller_file = module_referrer[:module_referrer.rindex("#")]
+                    caller_file_path = f"/{os.path.relpath(abs_caller_file, root_folder)}"
+
+                    try:
+                        caller_context = dpath.get(definition_context[abs_caller_file],
+                                                   # HACK ALERT: module data is currently double-nested in
+                                                   #             definition context. If fixed, remove the
+                                                   #             addition of "module." at the beginning.
+                                                   "module." + referrer_id,
+                                                   separator=".")
+                    except KeyError:
```

Comment:
is thisi expected? do we want to log in debug such case?

Reply:
Yes, good point. I’ll add that. One of the parser test cases hits that branch so I know it happens though I haven’t dug in to why.

Reply:
Done